### DISCUSSION TITLE
Add support for appimage

### DISCUSSION THREAD

**Discussion by jnweiger** [linuxmint/discussions#188] [Ideas]
Seen with nemo 5.6.3 on Linux Mint

* User downloads an appimage through e.g. firefox.
* Firefox shows the finished download and the user clicks on it, in order to run the appimage.
* A window named Library (not sure if that is from nautilus or nemo or unrelated) shows
![grafik](https://github.com/linuxmint/discussions/assets/1108546/10f13362-2da3-444f-a7e6-2f46963d371b)
----
* User launches nemo, and navigates to the Downloads folder.
* nemo also does not offer anything useful:
![grafik](https://github.com/linuxmint/discussions/assets/1108546/8f8d1488-ecc2-4479-80f7-bbfa70d4c14a)
----

Expected behaviour:
* appimage is a well known binary format nowadays.
* Instead of "Choose a program", nemo should offer "Make executable"  and maybe remind the user about trusted and untrusted sources.
* Once the user clicks on Make executable, the button could change to "Start appimage" and maybe with a reminder to move the appimage to better location.

I understand that nemo's current behaviour is in sync with other file browsers like nautilus.
But I don't understand, why that obvious (and probably simple) enhancement is not discussed more.

**Comment by jnweiger** [comment]
Further enhancement ideas:
* use the icon embedded in the appimage, instead of a generic gear icon
* capture stdout/stderr, and when an appimage quits immediately, display a popup. A common first time user issue is 
```
dlopen(): error loading libfuse.so.2

AppImages require FUSE to run. 
You might still be able to extract the contents of this AppImage 
if you run it with the --appimage-extract option. 
See https://github.com/AppImage/AppImageKit/wiki/FUSE 
for more information
```

**Reply by mtwebster** [reply]
fyi, you can install `xapp-appimage-thumbnailer` for icons.

**Reply by jnweiger** [reply]
Thanks for that hint. I found an installable package at https://github.com/mttbernardini/appimage-thumbnailer/releases
Hmm, seems it chooses the lowest possible resolution, but better than nothing:
![grafik](https://github.com/linuxmint/discussions/assets/1108546/da0c8b5a-9c96-4086-a0b7-9bcd3cbeaa0d)

**Reply by mtwebster** [reply]
I'm not familiar with that one.  Is xapp-appimage-thumbnailer not installable?
On Mint 21,
```
apt install xapp-appimage-thumbnailer
```

**Comment by jnweiger** [comment]
Here is how openSUSE Leap 15.4 (KDE desktop) does it:

![grafik](https://github.com/linuxmint/discussions/assets/1108546/b0055fba-2ece-43ab-a447-b99c45439151)
----

**Reply by Jeremy7701** [reply]
I would not want appimages (or indeed flatpacks) polluting a LMDE installation, either deliberately or as a result of some automatic upgrade process.
It would need to be something that was disabled by default.

**Reply by jnweiger** [reply]
I am not familiar with the differences between LMDE and "good old" Linux Mint. For me Mint has been a distribution with a good focus on the kind of usability that is needed in my environment. I recommend Mint to students a lot. I would not mind to have helpful usability disabled by default, as long as there is a hint (if that path gets approval).

I don't want to turn this into a political discussion, but I don't see an appimage as pollution these days.
In the past it was a horrible thought to have many copies of (slowly aging) libraries hidden in appimage. But in my experience that is not a big issue any more. Updated appimages are often released quickly (e.g. directly at github).
Upgrade / downgrade / uninstall are so transparent and straightfoward, that as an end-user one can develop a good trust in the reliability of an appimage.

**Reply by Jeremy7701** [reply]
**Personally**, I trust software from Debian Stable and Mint (where written for LMDE), but I couldn't have the same degree of trust for some (probably Ubuntu) application enclosed in a app image etc.
Since I'm running a basically Debian system, I'd consider it safe to install Debian packages - but very **un-safe** to install Ubuntu packages.
But different people are obviously going to have different viewpoints.
Functionally, Mint and LMDE are virtually identical, except that Mint is built on a Ubuntu base whilst LMDE is built on a Debian base.

**Reply by JustTryingToGetSomeWorkDone** [reply]
Jeremy, only you are in control of your fingers. Simply do not install software you don't want to install. No one is forcing you to do anything. Nor is anyone suggesting you be forced to do anything.

**Comment by edgarogh** [comment]
Great suggestion, this has bothered me for a few years!

This should actually be implemented for all ELF executables too really. `.run` is also a pretty common file extension for programs that are supposed to be chmoded and run as-is. If an ELF header is found, the option to launch as-is should be suggested.

I've never ever contributed to any of Mint's software, but if this sounds like a not-too-unapproachable first issue, I'm willing to give it a try.

![image](https://github.com/linuxmint/discussions/assets/46636609/c6c339f7-25cd-4eca-8d4c-d30f36a506b0)

**Comment by AtomicRobotMan0101** [comment]
Came here today to make this EXACT recommendation.

This week Ive migrated one noob off Windows and another off MacOS and both were stuck after downloading an appImage.

By having a trivial launcher which asks the user WHAT they'd like to do, for this very specific thing, seems like a bloody good idea.

**Comment by JustTryingToGetSomeWorkDone** [comment]
I would be content with just having the ability to automatically use the AppImage's embedded icon without having to extract the entire AppImage to get it. It seems very archaic and anti-user to be able to see the AppImage's correct icon in Nemo but not be able to get a menu item to do the same. Full AppImage integration would be very much appreciated. No one is forcing anyone to download AppImages, but some software is ONLY available via AppImage. Not supporting AppImage will cause some users to pass on Mint so they can use applications they deem essential.

**Comment by AtomicRobotMan0101** [comment]
I admin a lot of systems now. Ive been putting this project onto all my Mint installs. 

People love it - simply because it recreates their macOS experience. Click, it opens, drag to install, done.

https://github.com/kem-a/AppManager

This would be worthy to include in Mint on the next release.

**Reply by JustTryingToGetSomeWorkDone** [reply]
Thank you VERY much for this recommendation. Barring similar functionality being built in, there's no reason to use any other method.
